packages/loopring_v3/contracts/core/impl/ExchangeV3.sol

```diff
@@ -548,30 +548,36 @@ contract ExchangeV3 is IExchangeV3, ReentrancyGuard
 
     function approveTransaction(
         address owner,
-        bytes32 transactionHash
+        bytes32 transactionHash,
+        ExchangeData.ApprovalType approvalType
         )
         external
         override
         nonReentrant
         onlyFromUserOrAgent(owner)
     {
         state.approvedTx[owner][transactionHash] = true;
-        emit TransactionApproved(owner, transactionHash);
+        emit TransactionApproved(owner, transactionHash, approvalType);
```

Comment:
There's no way to check if the `approvalType` value specified is actually the expected value, so I don't really see how it can be used in a reliable way like this.